src/java_bytecode/library/src/java/lang/Class.java

```diff
@@ -0,0 +1,423 @@
+            "java.lang.Short".equals(name) ||
+            "java.lang.Integer".equals(name) ||
+            "java.lang.Long".equals(name) ||
+            "java.lang.Float".equals(name) ||
+            "java.lang.Double".equals(name) ||
+            "java.lang.Void".equals(name);}
+
+    public boolean isAnnotation() { return isAnnotation; }
+    public boolean isSynthetic() { return isSynthetic; }
+    public boolean isLocalClass() { return isLocalClass; }
+    public boolean isMemberClass() { return isMemberClass; }
+    public boolean isAnonymousClass() { return "".equals(getSimpleName()); }
+    public boolean isEnum() { return isEnum; }
+    private boolean isLocalOrAnonymousClass() {
+        return isLocalClass() || isAnonymousClass();
+    }
+
+    public String getName() {
+        // TODO: this is only for objects, and primitive types and arrays need
+        // a special treatment
+        return this.name;
+    }
+
+    /**
+     * Returns the class loader for the class.  Some implementations may use
+     * null to represent the bootstrap class loader. This method will return
+     * null in such implementations if this class was loaded by the bootstrap
+     * class loader.
+     *
+     * <p> If a security manager is present, and the caller's class loader is
+     * not null and the caller's class loader is not the same as or an ancestor of
+     * the class loader for the class whose class loader is requested, then
+     * this method calls the security manager's {@code checkPermission}
+     * method with a {@code RuntimePermission("getClassLoader")}
+     * permission to ensure it's ok to access the class loader for the class.
+     *
+     * <p>If this object
+     * represents a primitive type or void, null is returned.
+     *
+     * @return  the class loader that loaded the class or interface
+     *          represented by this object.
+     * @throws SecurityException
+     *    if a security manager exists and its
+     *    {@code checkPermission} method denies
+     *    access to the class loader for the class.
+     * @see java.lang.ClassLoader
+     * @see SecurityManager#checkPermission
+     * @see java.lang.RuntimePermission
+     */
+    @CallerSensitive
+    public ClassLoader getClassLoader() {
+        ClassLoader cl = getClassLoader0();
+        if (cl == null)
+            return null;
+        SecurityManager sm = System.getSecurityManager();
+        if (sm != null) {
+            ClassLoader.checkClassLoaderPermission(cl, Reflection.getCallerClass());
+        }
+        return cl;
+    }
+
+    ClassLoader getClassLoader0() {
+        // DIFFBLUE MODEL LIBRARY The real java.lang.Class stores a
+        // `private final ClassLoader classLoader` which is initialised by the
+        // jvm rather than by the constructor of this object.
+        // TODO test-gen should understand this method natively.
+        return null;
+    }
+
+    public String getSimpleName() {
```

Comment:
The documentation of some classes is missing (as it is probably in the model library). I recommend adding back the original documentation from the JDK so that a javadoc can be generated, Also, any CProver-specific comments could be added as custom Java tags (such as the `@diffblue.fullSupport` we use in the model library).

Reply:
Same comment, this is a copy of `model/src/main/java/java/lang/Class.java`. Are you sure we should do this change here?